content/blog/eng/automation-testing-a-step-back.md

```diff
@@ -0,0 +1,114 @@
+
+That said a code is "better" than another when, even if does the same things, it is easier to maintain and to evolve (being more readable, respecting the SOLID principles, etc...).
+
+Tests help to increase quality, because making testable code forces us to restructure it and to change dependency management. This generally translates into a more correct distribution of responsibilities. So it's better from an architectural point of view.
+
+This doesn't mean that untested code have worse distribution of responsibilities. But testing can help to achieve this goal putting us in a position where we need to rethink our code structure.
+
+## Find bug during development
+
+I think it's uncommon but can happen to find a bug out of an automated test during development. It happens because usually the tested cases in automated tests are more structured than manual tests. Precisely because we are forced to write them and we know that they will remain written, it happens that we pay more attention to them. A developer rarely writes all the manual test cases to be performed and then runs them based on how he wrote them. For this reason, the process of writing an automatic test spots problems that may go unnoticed during a manual test.
+
+Contrary to the first two points, the benefits of this last two are immediate and do not increase over time. So a strategy could be to delete the test after the development is done. I know that it sounds weird but it makes sense if we don't consider the other points.
+
+## Honorable mentions
+
+There is another point that usually people talk about as a benefit of working on a well-tested project: it is confidence; since developing while having your shoulders covered by tests is just mentally lighter.
+
+I didn't put it on the list because I am not comparing automation-tests with no-tests, but automation-tests with manual-tests.
+
+# Can test harm?
+
+We have seen what are the pros of automation testing, now let's examine what are the cons.
+
+Tests come at a cost: they can slow down the development process, they can slow down the CI, they can make harder maintenance and refactoring, they can make you write worst code.
+
+## Slow down the development process
+
+It has happened to me several times that writing tests has consumed me much more time and effort than developing the tested feature. It can happen due to inexperience, or due to technical limitations for which there is a greater complexity in writing tests.
+
+In these cases I always ask myself if it is really worth it, if the time invested in writing these tests will pay off, and if so after how long.
+
+So focusing only on the first of the benefits, for simplicity, we compare the time invested by writing the tests with the gain that will be obtained over time. When investing money, we can understand if the investment is good based on how long it will take to get back into the investment. The same rule applies to software automation.
```

Comment:
Therefore, focusing only on the first of the benefits for simplicity'sake, we compare the time spent in writing the tests with the gain that will be obtained over time. When you invest money, you can see if the investment is good based on how long it will take to get back into the investment.